extensions/battleroyale/standalone.mjs

```diff
@@ -0,0 +1,532 @@
+        this.bullets.delete(bulletID);
+        await this.helper.broadcastCellSetUpdateToAllUser(
+            'update',
+            mapCoord.mapName,
+            CellSet.fromObject({
+              name: LAYER_BULLET.layerName,
+              cells: Array.from(
+                  this.bullets,
+                  ([_, v]) => v['mapCoord'],
+              ),
+            }),
+        );
+      } else {
+        const players = this.helper.gameState.getPlayers();
+        const playerMap = new Map();
+        players.forEach((player, playerID) => {
+          const {world, x, y} = player.mapCoord;
+          playerMap.set(String(world)+'#'+String(x)+'#'+String(y), playerID);
+        });
+        for (const [bulletID, bullet] of this.bullets) {
+          const {world, x, y} = bullet.mapCoord;
+          const str = String(world)+'#'+String(x)+'#'+String(y);
+          if (playerMap.has(str)) {
+            this.killPlayer(playerMap.get(str));
+            bullet.deleteInterval();
+            this.bullets.delete(bulletID);
+            playerMap.delete(bullet.mapCoord);
+          }
+        }
+        await this.helper.broadcastCellSetUpdateToAllUser(
+            'update',
+            mapCoord.mapName,
+            CellSet.fromObject({
+              name: LAYER_BULLET.layerName,
+              cells: Array.from(
+                  this.bullets,
+                  ([_, v]) => v['mapCoord'],
+              ),
+            }),
+        );
+      }
+    }, BULLET_SPEED, bulletID);
+
+
+    return true;
+  }
+  /**
+   * Given a gameMap return all edge
+   * @param {Set} fires the fire Set
+   * @param {Map} cellSets the maps
+   * @param {Number} fireCounter count edge width
+   * @return {Boolean} need further update or not
+   */
+  updateEdge(fires, cellSets, fireCounter) {
+    let noUpdateCounter = 0; let totalCell = 0;
+    for (const {cells} of cellSets) {
+      for (const {x, y, w, h} of cells) {
+        const wbound = Math.min(fireCounter, Math.floor(w/2));
+        const hbound = Math.min(fireCounter, Math.floor(h/2));
+        totalCell++;
+        if (wbound === Math.floor(w/2) && hbound === Math.floor(h/2)) {
+          noUpdateCounter++;
+        }
+        fires.add({x, y, w, h: hbound});
+        fires.add({x, y: y+h-hbound, w, h: hbound});
+        fires.add({x, y, w: wbound, h});
+        fires.add({x: x+w-wbound, y, w: wbound, h});
+      }
+    }
+    return noUpdateCounter != totalCell;
+  }
+
+  /**
+   *
+   * @param {Object} msg
+   * @return {undefined}
+   */
+  async onPlayerUpdate(msg) {
+    const {mapCoord, playerID} = msg;
+
+    if (typeof playerID !== 'string') {
+      console.warn('Invalid PlayerSyncMessage with non-string msg.playerID', msg, msg.playerID);
+      return;
+    }
+    if (mapCoord === 'undefined') {
+      // No location? Just skip.
+      return;
+    }
+
+    const bullets = this.helper.gameMap.getCell(mapCoord, LAYER_BULLET.layerName);
+    if (bullets) {
+      this.killPlayer(playerID);
```

Comment:
Will a player be killed more than once?
If not, please write down the reason somewhere; otherwise, please make sure that this does no harm and write down the clarification somewhere.